vars/checkoutGitRepo.groovy

```diff
@@ -28,17 +47,16 @@ def call(body) {
     body.delegate = config
     body()
 
-    def backupBranch = config.backupBranch  ?: false
+    def isBackupConfigured = config.backupBranch ?: false
 
     try {
-      checkoutWithConfig(config)
+        checkoutWithConfig(config)
     } catch (Exception e) {
-        if (backupBranch) {
-          config.branch = backupBranch
-          checkoutWithConfig(config)
+        if (isBackupConfigured) {
+            config.branch = 'backupBranch'
```

Comment:
Why does this change to a static string of `'backupBranch'`, rather than the value of `config.backupBranch`?

Reply:
Need to change it back :)

Reply:
just found that little bit awkward as backupBranch is String and boolean at same time :). Changed it back

Reply:
Yeah, you're right that the 'quantum typing' of the `backupBranch` variable is not very nice. How about the best of both worlds?

```groovy
final Boolean isBackupConfigured = config.backupBranch ? true : false

// ...

if (isBackupConfigured) {
    config.branch = config.backupBranch
}
```

Reply:
@grdryn I think we can leave it as it is/Its readable but for some newbies should be confused :)